app/javascript/src/lib/OWLanguageLinter.js

```diff
@@ -569,7 +569,7 @@ function findEventBlocksWithMissingArguments(content) {
 
     const requiredEventArgs = eventTypeToArgumentsMap[eventType.toLowerCase()]
 
-    if (requiredEventArgs.length !== givenEventArgs.length) {
+    if (requiredEventArgs && requiredEventArgs.length !== givenEventArgs.length) {
```

Comment:
Let's use optional chaining
```suggestion
    if (requiredEventArgs?.length !== givenEventArgs.length) {
```

Reply:
❌ Bad.
If `requiredEventArgs == null` and `givenEventArgs = [...]`, then `requiredEventArgs?.length = null`, then `requiredEventArgs?.length !== givenEventArgs.length => null !== [...] => true`.

Optional chaining here means we always go into the if block when we don't have a record of the event type.

Reply:
Sorry I have no idea what you mean